File: java/Algorithms/BinarySearch.java
package Algorithms;

public class BinarySearch {
    public static int binarySearch(int[] numbers, int low, int high, int target) {
        // base case
        if (low <= high) {
            // calculate middle index
            int mid = (low + high)/2;
            
            if (target == numbers[mid]) {
                return mid;
            } else if (target < numbers[mid]) {
                return binarySearch(numbers, low, mid - 1, target);
            } else { // target > numbers[mid]
                return binarySearch(numbers, mid + 1, high, target);
            }
        } else {
            return -1;
        }
    }
}

File: python/binary-search.py
def binarySearch(nums, low, high, target):
    # Base Case
    if low <= high:
        # Calculate middle index
        mid = int((low + high)/2)

        if target == nums[mid]:
            return mid
        elif target < nums[mid]:
            return binarySearch(nums, low, mid - 1, target)
        else: # target > nums[mid]:
            return binarySearch(nums, mid + 1, high, target)
    else:
        return -1
